src/server/macos.rs

```diff
@@ -147,140 +157,157 @@ impl<'os> Method for PackageMethod<'os, Macos> {
-                    cmd, out.status);
-            }
-            let mut result = Vec::new();
-            let lines = out.stdout.split(|&b| b == b'\n')
-                .filter_map(|line| str::from_utf8(line).ok());
-            for line in lines {
-                if !line.starts_with("com.edgedb.edgedb-server-") {
-                    continue;
-                }
-                let major = &line["com.edgedb.edgedb-server-".len()..].trim();
-
+        Ok(self
+            .installed
+            .get_or_try_init(|| {
                 let mut cmd = StdCommand::new("pkgutil");
-                cmd.arg("--pkg-info").arg(line.trim());
-                let out = cmd.output()
-                    .context("cannot get package version")?;
-                if !out.status.success() {
-                    anyhow::bail!("cannot get package version: {:?} {}",
-                        cmd, out.status);
+                cmd.arg(r"--pkgs=com.edgedb.edgedb-server-\d.*");
+                let out = cmd.output().context("cannot get installed packages")?;
+                if out.status.code() == Some(1) {
+                    return Ok(Vec::new());
+                } else if !out.status.success() {
+                    anyhow::bail!("cannot get installed packages: {:?} {}", cmd, out.status);
                 }
-                let lines = out.stdout.split(|&b| b == b'\n')
+                let mut result = Vec::new();
+                let lines = out
+                    .stdout
+                    .split(|&b| b == b'\n')
                     .filter_map(|line| str::from_utf8(line).ok());
-                let mut version = None;
                 for line in lines {
-                    if line.starts_with("version:") {
-                        version = Some(line["version:".len()..].trim());
+                    if !line.starts_with("com.edgedb.edgedb-server-") {
+                        continue;
+                    }
+                    let major = &line["com.edgedb.edgedb-server-".len()..].trim();
+
+                    let mut cmd = StdCommand::new("pkgutil");
+                    cmd.arg("--pkg-info").arg(line.trim());
+                    let out = cmd.output().context("cannot get package version")?;
+                    if !out.status.success() {
+                        anyhow::bail!("cannot get package version: {:?} {}", cmd, out.status);
+                    }
+                    let lines = out
+                        .stdout
+                        .split(|&b| b == b'\n')
+                        .filter_map(|line| str::from_utf8(line).ok());
+                    let mut maybe_version: Option<&str> = None;
+                    for line in lines {
+                        maybe_version = match line.strip_prefix("version:") {
+                            Some(version) => Some(version),
+                            None => continue,
+                        };
+
                         break;
```

Comment:
This is probably not the most elegant solution but I'm new to Rust and I tried to stay close to existing logic.
`break VALUE;` is not possible in for loop.